pkg/drivers/kic/oci/network_create.go

```diff
@@ -0,0 +1,197 @@
+		return gateway, nil
+	}
+	// simple way to create networks, subnet is taken, try one bigger
+	attempt := 0
+	subnetAddr := defaultSubnetAddr
+	gateway, err = tryCreateDockerNetwork(subnetAddr, defaultSubnetRange, clusterName)
+	if err != nil {
+		if err != ErrNetworkSubnetTaken {
+			return nil, errors.Wrapf(err, "error creating network")
+		}
+		// try up to 13 times
+		// we can try up to 255
+		for attempt < 13 {
+			attempt++
+			glog.Infof("Couldn't create network %q at %q subnet will try again with a new subnet ...", clusterName, subnetAddr)
+			// increase 3nd digit by 10 each time
+			// 13 times adding 10 defaultSubnet "192.168.39.0/24"
+			// at most it will add up to 169 which is still less than max allowed 255
+			// this is large enough to try more and not too small to not try enough
+			// can be tuned in the next iterations
+			ip := net.ParseIP(subnetAddr).To4()
+			ip[2] += byte(9 + attempt)
+
+			gateway, err = tryCreateDockerNetwork(ip.String(), defaultSubnetRange, clusterName)
+			if err == nil {
+				return gateway, nil
+			}
+			if errors.Is(err, ErrNetworkSubnetTaken) || errors.Is(err, ErrNetworkGatewayTaken) {
+				continue
+			}
+		}
+
+	}
+	return gateway, nil
+}
+
+func tryCreateDockerNetwork(subnetAddr string, subnetRange int, name string) (net.IP, error) {
+	gateway := net.ParseIP(subnetAddr)
+	gateway.To4()[3]++ // first ip for gateway
+	glog.Infof("attempt to create network %q with subnet: %s and gateway %s...", subnetAddr, name, gateway)
+	// options documentation https://docs.docker.com/engine/reference/commandline/network_create/#bridge-driver-options
+	rr, err := runCmd(exec.Command(Docker, "network", "create", "--driver=bridge", fmt.Sprintf("--subnet=%s", fmt.Sprintf("%s/%d", subnetAddr, subnetRange)), fmt.Sprintf("--gateway=%s", gateway), "-o", "--ip-masq", "-o", "--icc", fmt.Sprintf("--label=%s=%s", CreatedByLabelKey, "true"), name))
+	if err != nil {
+		if strings.Contains(rr.Output(), "Pool overlaps with other one on this address space") {
+			return nil, ErrNetworkSubnetTaken
+		}
+		if strings.Contains(rr.Output(), "failed to allocate gateway") && strings.Contains(rr.Output(), "Address already in use") {
+			return nil, ErrNetworkGatewayTaken
+		}
+		return nil, errors.Wrapf(err, "error creating network")
+	}
+	return gateway, nil
+}
+
+// returns subnet and gate if exists
+func dockerNetworkInspect(name string) (*net.IPNet, net.IP, error) {
+	rr, err := runCmd(exec.Command(Docker, "network", "inspect", name, "--format", "{{(index .IPAM.Config 0).Subnet}},{{(index .IPAM.Config 0).Gateway}}"))
+	if err != nil {
+		if strings.Contains(rr.Output(), "No such network:") {
+			return nil, nil, ErrNetworkNotFound
+		}
+		return nil, nil, err
+	}
+	// results looks like 172.17.0.0/16,172.17.0.1
+	ips := strings.Split(strings.TrimSpace(rr.Stdout.String()), ",")
+	if len(ips) == 0 {
+		return nil, nil, fmt.Errorf("invalid network info")
```

Comment:
Include the value of rr.Stdout.String() in the error, because this will be painful to debug otherwise.

Could you also make the message more precise? Perhaps `empty IP list parsed from %q`